pkg/client/cache/ns_index_listers.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+const (
+	NameSpaceIndex string = "namespace"
+)
+
+// IndexerToPodLister makes a Store have the List method of the client.PodInterface
+// The Store must contain (only) Pods, and just expects "namespace" index.
+type IndexerToPodLister struct {
```

Comment:
Should this object be renamed if it's specific to namespaces?

Maybe `PodListerByNamespace` or something?